components/playlist/browser/media_detector_component_manager.cc

```diff
@@ -138,7 +148,11 @@ void MediaDetectorComponentManager::OnGetScripts(
     return;
   }
 
-  DCHECK(script_map.count(GetBaseScriptName()));
+  CHECK(script_map.contains(GetMediaSourceAPISuppressorScriptName()));
```

Comment:
I'm not sure if this should be a `CHECK`? Do we want to crash if there's some mistake in the crx?

Reply:
`script_map` is non-empty at this point, so I think so.

Reply:
but what I'm saying is if there's some screwup with the crx and it doesn't contain the file we expect, crashing doesn't seem ideal. I think we would just bail out of the update instead?

Reply:
Creating a follow-up issue for that. Also, these changes don't change the current behavior, as `std::map<>::at()` throws `std::out_of_range` if the specific key is not in the map.

Reply:
https://github.com/brave/brave-browser/issues/36445